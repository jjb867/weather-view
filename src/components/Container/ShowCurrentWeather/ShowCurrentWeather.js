import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, GetWeatherAnimation } from '../../../utils/poseAnimations';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { isWeatherResultEmpty } from '../../../utils/weatherUtil';
import WeatherCard from '../../WeatherCard/WeatherCard';

class ShowCurrentWeather extends Component {

    componentDidMount() {
        const { currentWeather } = this.props.weather;
        if (isWeatherResultEmpty(currentWeather)) {
            this.props.history.push('/');
        }
    }

    onKeyPress = e => {
        if (e.key === 'Enter') {
            this.props.history.push('/');
        }
    };

    onBackToSearchClick = () => {
        this.props.history.push('/');
    };

    render() {
        const { currentWeather } = this.props.weather;
        const currentWeatherEmpty = isWeatherResultEmpty(currentWeather);

        return (
            <React.Fragment>
                {!currentWeatherEmpty && (
                    <GetWeatherAnimation key="search" className="search" initialPose="hidden" pose="visible">
                        <div className="container" onKeyPress={this.onKeyPress}>
                            <section className="section">
                                <Button className="button is-small is-link" onClick={this.onBackToSearchClick}>
                                    <FontAwesomeIcon icon="arrow-left" />
                                    &nbsp;Back to Search
                                </Button>
                                <WeatherCard weather={currentWeather} />
                            </section>
                        </div>
                    </GetWeatherAnimation>
                )}
            </React.Fragment>
        );
    }
}

const mapStateToProps = state => ({
    weather: state.weather,
    error: state.error,
});

ShowCurrentWeather.propTypes = {
    weather: PropTypes.object.isRequired,
};

export default connect(
    mapStateToProps,
    {},
)(ShowCurrentWeather);
