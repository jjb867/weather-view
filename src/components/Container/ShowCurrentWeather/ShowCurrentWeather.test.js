import React from 'react';
import { shallow, mount } from 'enzyme';
import ShowCurrentWeather from './ShowCurrentWeather';

describe('ShowCurrentWeather', () => {
    it('should render without crashing', () => {
        const component = shallow(<ShowCurrentWeather/>);
        expect(component).toMatchSnapshot();
    });
    it('should redirect if the weather object is empty', () => {

    });
    describe('weather object is empty', () => {
        it('should redirect to get weather', () => {

        });
    });
});