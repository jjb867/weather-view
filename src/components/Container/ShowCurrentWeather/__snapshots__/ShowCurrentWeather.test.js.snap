// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCurrentWeather should render without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
